src/main/java/lt/velykis/roberta/budget/app/transaction/TransactionRestController.java

```diff
@@ -0,0 +1,107 @@
+package lt.velykis.roberta.budget.app.transaction;
+
+import lt.velykis.roberta.budget.app.account.Account;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/rest/transactions")
+public class TransactionRestController {
+
+    private final TransactionService transactionService;
+
+    @Autowired
+    public TransactionRestController(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    @GetMapping("")
+    public TransactionsWithTotalAmount listAllTransactions(@RequestParam(required = false, name = "accountId") Optional<UUID> accountId) {
+
+        List<Transaction> transactions = accountId.map(transactionService::filterTransaction)
+                .orElseGet(transactionService::getAllTransactions);
+
+        BigDecimal totalAmount = TransactionService.countTotal(transactions);
+        return new TransactionsWithTotalAmount(transactions, totalAmount);
+
+    }
+
+    @GetMapping("/{transactionId}")
+    public Optional<Transaction> transaction(@PathVariable("transactionId") UUID transactionId) {
+
+        Optional<Transaction> transaction = transactionService.findTransaction(transactionId);
+
+        if (!transaction.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Transaction not found");
+        }
+
+        return transactionService.findTransaction(transactionId);
+    }
+
+    @PostMapping("")
+    public List<Transaction> addTransaction(@Valid @RequestBody Transaction transaction,
+                                            BindingResult bindingResult) {
+
+        if (bindingResult.hasErrors()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, String.valueOf(bindingResult.getAllErrors()));
+        }
+
+        Optional<Account> account = transactionService.findAccount(transaction.getAccountId());
+
+        if (!account.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Account not found");
```

Comment:
if account not found, I'd say is a BAD_REQUEST (i.e. someone wants to create a transaction with bad accountId), because NOT_FOUND should be for when the transaction itself is not found. NOT_FOUND would maybe be suitable if the accountId was in the path, not in the payload